File: CSCD212/CommandPattern/src/DocumentWriteCommand.java
public class DocumentWriteCommand implements Command {
    private Document document;
    private String   text;

    /**
     * @param document Document object passed from DocumentInvoker class.
     * @param text     Text to be added to the document.
     */
    public DocumentWriteCommand(Document document, String text) {
        this.document = document;
        this.text     = text;
        document.Write(text);
    }

    @Override
    public void undo() {
        document.Erase(text);
    }

    @Override
    public void redo() {
        document.Write(text);
    }
}

File: Personal/CommandPattern/pkg/document_write_command.py
from pkg.command import Command


class DocumentWriteCommand(Command):
    """Write 'text' to the document."""

    def __init__(self, document, text):
        """Constructor.

        Parameters
        ----------
        document : document.Document
            Document object/instance.
        text : str
            Text to be added to the document.
        """
        self._document = document
        self._text = text
        self._document.write(text)

    def undo(self):
        """Erase 'text' from the document."""
        self._document.erase(self._text)

    def redo(self):
        """Re-add 'text' to the document."""
        self._document.write(self._text)
